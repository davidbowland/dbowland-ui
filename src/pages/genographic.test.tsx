import GenographicInfographic from '@components/genographic-infographic'
import PrivacyLink from '@components/privacy-link'
import TitleBar from '@components/title-bar'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import React from 'react'

import Genographic, { Head } from './genographic'

jest.mock('@components/genographic-infographic')
jest.mock('@components/privacy-link')
jest.mock('@components/title-bar')

describe('Genographic page', () => {
  beforeAll(() => {
    jest.mocked(GenographicInfographic).mockReturnValue(<></>)
    jest.mocked(PrivacyLink).mockReturnValue(<></>)
    jest.mocked(TitleBar).mockReturnValue(<></>)
  })

  it('should render GenographicInfographic', () => {
    render(<Genographic />)
    expect(GenographicInfographic).toHaveBeenCalledTimes(1)
  })

  it('should render TitleBar', () => {
    render(<Genographic />)
    expect(TitleBar).toHaveBeenCalledTimes(1)
  })

  it('renders Head', () => {
    render(<Head />)
    expect(document.title).toEqual('Genographic Information | dbowland.com')
  })
})
