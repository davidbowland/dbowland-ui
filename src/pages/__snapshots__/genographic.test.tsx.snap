// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genographic page Ensure snapshot match 1`] = `
<main>
  <p>
    In 2016, I sent my DNA to the now-defunct National Geographic Genographic project to both help drive research and to learn something about the immense journey that brought me here. Below are the results from my participation in that project. People who know me will not be surprised to learn that my ancestors were "[h]unters of megafauna such as the woolly mammoth." ðŸ˜†
  </p>
  <figure
    class="sc-bdnxRM lpgUIQ"
  >
    <a
      href="test-file-stub"
    />
    <figcaption
      class="sc-gtsrHT bsjUbs"
    >
      Click or tap on the image for the full 18-page PDF report.
    </figcaption>
  </figure>
</main>
`;
